contrib/babelfishpg_tsql/src/pltsql_ruleutils.c

```diff
@@ -369,6 +376,179 @@ tsql_get_constraintdef(PG_FUNCTION_ARGS)
+	appendStringInfo(&buf, "CREATE %s %s(",
+					 isfunction ? "FUNCTION" : "PROCEDURE",
+					 tsql_quote_qualified_identifier(nnsp, name));
+	tmp = SysCacheGetAttr(PROCOID, proctup, Anum_pg_proc_probin, &isnull);
+        number_args=proc->pronargs; 
+        if(isfunction) number_args++;
+       	probin_json_reader(tmp, &typmod_arr, number_args);
+	(void) print_function_arguments(&buf, proctup, false, true, &typmod_arr);
+	appendStringInfoString(&buf, ")");
+	if (isfunction)
+	{
+		appendStringInfoString(&buf, " RETURNS ");
+		print_function_rettype(&buf, proctup, &typmod_arr, number_args);
+	}
+
+	/* Emit some miscellaneous options on one line */
+	oldlen = buf.len;
+
+        if (proc->proisstrict)
+		appendStringInfoString(&buf, " WITH RETURNS NULL ON NULL INPUT");
+
+	/* Emit any proconfig options, one per line */
+	if (!isnull)
+	{
+		ArrayType  *a = DatumGetArrayTypeP(tmp);
+		int			i;
+
+		Assert(ARR_ELEMTYPE(a) == TEXTOID);
+		Assert(ARR_NDIM(a) == 1);
+		Assert(ARR_LBOUND(a)[0] == 1);
+
+		for (i = 1; i <= ARR_DIMS(a)[0]; i++)
+		{
+			Datum		d;
+
+			d = array_ref(a, 1, &i,
+						  -1 /* varlenarray */ ,
+						  -1 /* TEXT's typlen */ ,
+						  false /* TEXT's typbyval */ ,
+						  TYPALIGN_INT /* TEXT's typalign */ ,
+						  &isnull);
+			if (!isnull)
+			{
+				char	   *configitem = TextDatumGetCString(d);
+				char	   *pos;
+
+				pos = strchr(configitem, '=');
+				if (pos == NULL)
+					continue;
+				*pos++ = '\0';
+
+				appendStringInfo(&buf, " SET %s TO ",
+								 tsql_quote_identifier(configitem));
+
+				/*
+				 * Variables that are marked GUC_LIST_QUOTE were already fully
+				 * quoted by flatten_set_variable_args() before they were put
+				 * into the proconfig array.  However, because the quoting
+				 * rules used there aren't exactly like SQL's, we have to
+				 * break the list value apart and then quote the elements as
+				 * string literals.  (The elements may be double-quoted as-is,
+				 * but we can't just feed them to the SQL parser; it would do
+				 * the wrong thing with elements that are zero-length or
+				 * longer than NAMEDATALEN.)
+				 *
+				 * Variables that are not so marked should just be emitted as
+				 * simple string literals.  If the variable is not known to
+				 * guc.c, we'll do that; this makes it unsafe to use
+				 * GUC_LIST_QUOTE for extension variables.
+				 */
+				if (GetConfigOptionFlags(configitem, true) & GUC_LIST_QUOTE)
+				{
+					List	   *namelist;
+					ListCell   *lc;
+
+					/* Parse string into list of identifiers */
+					if (!SplitGUCList(pos, ',', &namelist))
+					{
+						/* this shouldn't fail really */
+						elog(ERROR, "invalid list syntax in proconfig item");
+					}
+					foreach(lc, namelist)
+					{
+						char	   *curname = (char *) lfirst(lc);
+
+						simple_quote_literal(&buf, curname);
+						if (lnext(namelist, lc))
+							appendStringInfoString(&buf, ", ");
+					}
+				}
+				else
+					simple_quote_literal(&buf, pos);
+			}
+		}
+	}
```

Comment:
At this point, tmp holds the value of probin from pg_proc catalog not proconfig which contradicts to comment. Either we should remove it if de-parsing of proconfig is not needed for TSQL **OR** we've to fetch the value of proconfig before using it in this code.